metricflow/model/parsing/dir_to_model.py

```diff
@@ -134,11 +138,11 @@ def parse_config_yaml(
     data_source_class: Type[DataSource] = DataSource,
     metric_class: Type[Metric] = Metric,
     materialization_class: Type[Materialization] = Materialization,
-) -> List[Union[DataSource, Metric, Materialization]]:
+) -> Tuple[List[Union[DataSource, Metric, Materialization]], List[ValidationIssueType]]:
```

Comment:
Normally when you have a pair of lists in a return like this they line up. So I'd expect this to return a 1:1 mapping of results to issues, or possibly a result or an issue in a given index location.

But what this does is it interleaves them in the 1:1 case - if there's an issue I generally won't get a whatever - but it can also spawn a bunch of issues. This means there's no way for me to map issues back to whatever I'm seeing in the return output.

Would we be better off making this return a `List[Tuple[Union[DataSource etc.], [List[ValidationIssueType]]]]`? Then every entry has one or both of an object and a list of validation issues.

Alternatively, maybe we're better off with a more descriptive data structure that has `result` and `errors` and is documented as having one or the other but not both.

Reply:
It was not intended that the two lists be related in that manner. I.e. I wasn't intending to ensure a one-to-one matching. Additionally it's possible that an issue exists for an object that doesn't because creation of the object from the config failed. I'll go the route of having a more descriptive data structure that has `results` and `errors`, it'll be somewhat similar to `ModelBuildResults`